CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+set(CHPL_CMAKE_HOST_SYSTEM_LINK_ARGS CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_ARCH CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_CPU CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_COMPILER_PRGENV CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_BUNDLED_COMPILE_ARGS CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_SYSTEM_COMPILE_ARGS CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_BUNDLED_LINK_ARGS CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_SYSTEM_LINK_ARGS CACHE STRING "Description")
+set(CHPL_CMAKE_RUNTIME_CPU CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_CPU_FLAG CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_BACKEND_CPU CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_MEM CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_JEMALLOC CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_MEM CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_JEMALLOC CACHE STRING "Description")
+set(CHPL_CMAKE_MAKE CACHE STRING "Description")
+set(CHPL_CMAKE_GMP CACHE STRING "Description")
+set(CHPL_CMAKE_GMP_IS_OVERRIDDEN CACHE STRING "Description")
+set(CHPL_CMAKE_RE2_IS_OVERRIDDEN CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_SUPPORT CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_CONFIG CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_VERSION CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_CLANG_C CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_CLANG_CXX CACHE STRING "Description")
+set(CHPL_CMAKE_SANITIZE CACHE STRING "Description")
+set(CHPL_CMAKE_RUNTIME_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_LAUNCHER_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_COMPILER_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_BIN_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_BIN_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_SYS_MODULES_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_GASNET_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_GMP_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_HWLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_JEMALLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_JEMALLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_LIBFABRIC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_LIBUNWIND_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_QTHREAD_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_RE2_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_PE_CHPL_PKGCONFIG_LIBS CACHE STRING "Description")
+
+set(PRINTCHPLENV_COMMAND printchplenv --compiler --internal --cmake)
+# message(STATUS "running ${PRINTCHPLENV_COMMAND} from ${CMAKE_CURRENT_SOURCE_DIR}/util")
+
+find_package(Python)
+
+execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/util/${PRINTCHPLENV_COMMAND}
+                WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/util
+                TIMEOUT 5
```

Comment:
Is this 5 seconds? Probably 30 seconds would be more reasonable as a general timeout for printchplenv.